libraries/Microsoft.Bot.Builder.Prompts/DateTimePrompt.cs

```diff
@@ -1,86 +1,102 @@
-//using Microsoft.Recognizers.Text.DateTime;
-//using Microsoft.Recognizers.Text.Number;
-//using Microsoft.Recognizers.Text.NumberWithUnit;
-//using static Microsoft.Bot.Builder.Prompts.PromptValidatorEx;
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
 
-//namespace Microsoft.Bot.Builder.Prompts
-//{
-//    public class DateTimeOffsetResult
-//    {
-//        public DateTimeOffsetResult() { }
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Microsoft.Recognizers.Text;
+using Microsoft.Recognizers.Text.DateTime;
+using static Microsoft.Bot.Builder.Prompts.PromptValidatorEx;
 
-//        public DateTimeOffset Value { get; set; }
+namespace Microsoft.Bot.Builder.Prompts
+{
+    public class DateTimeResult : PromptResult
+    {
+        public string Text { get; set; }
+        public string Value { get; set; }
+        public string Start { get; set; }
+        public string End { get; set; }
+        public string Timex { get; set; }
+    }
 
-//        public string Text { get; set; }
-//    }
+    /// <summary>
+    /// CurrencyPrompt recognizes currency expressions as float type
+    /// </summary>
+    public class DateTimePrompt : BasePrompt<DateTimeResult>
+    {
+        private IModel _model;
 
-//    /// <summary>
-//    /// CurrencyPrompt recognizes currency expressions as float type
-//    /// </summary>
-//    public class DateTimePrompt : BasePrompt<DateTimeOffsetResult>
-//    {
-//        private IModel _model;
+        public DateTimePrompt(string culture, PromptValidator<DateTimeResult> validator = null)
+            : base(validator)
+        {
+            _model = new DateTimeRecognizer(culture).GetDateTimeModel();
+        }
 
-//        public DateTimePrompt(string culture, PromptValidator<DateTimeOffsetResult> validator = null)
-//            :base(validator)
-//        {
-//            _model = new DateTimeRecognizer(culture).GetDateTimeModel();
-//        }
+        protected DateTimePrompt(IModel model, PromptValidator<DateTimeResult> validator = null)
+            : base(validator)
+        {
+            _model = model;
+        }
 
-//        protected DateTimePrompt(IModel model, PromptValidator<DateTimeOffsetResult> validator = null)
-//            : base(validator)
-//        {
-//            this._model = model;
-//        }
+        /// <summary>
+        /// Used to validate the incoming text, expected on context.Request, is
+        /// valid according to the rules defined in the validation steps. 
+        /// </summary>        
+        public override async Task<DateTimeResult> Recognize(ITurnContext context)
+        {
+            BotAssert.ContextNotNull(context);
+            BotAssert.ActivityNotNull(context.Activity);
+            if (context.Activity.Type != ActivityTypes.Message)
+                throw new InvalidOperationException("No Message to Recognize");
```

Comment:
I wouldn't throw an error here.  I would just return not recognized.  It's quite possible for you to get called with a non-message activity and we don't want the developer to have to add tones of Activity.Type checks before they call every prompt.